### PR TITLE
feat: compile custom magic blocks

### PR DESCRIPTION
**Pull request by kellyjosephprice**
RM-3058 |
:---:|

## 🧰 Changes

Compiles custom magic blocks!

## 🧬 QA & Testing

- [ ] tests pass?

### CODE REVIEW

File: __tests__/__snapshots__/flavored-compilers.test.js.snap [hunk 1]
```diff
@@ -65,3 +65,16 @@ exports[`ReadMe Magic Blocks Figure 1`] = `
 [/block]
 "
 `;
+
+exports[`ReadMe Magic Blocks custom blocks 1`] = `
+"
+[block:undefined]
```
Comment (erunion) [inline __tests__/__snapshots__/flavored-compilers.test.js.snap:71]: undefined?

File: __tests__/magic-block-parser.test.js [hunk 2]
```diff
@@ -230,14 +228,11 @@ describe('Parse Magic Blocks', () => {
       "backgroundColor": "#ffffff",
       "title": "Tutorial Title",
       "emoji": "🦉",
-      "link": "http://test.com",
+      "link": "http://test.com"
```
Comment (erunion) [inline __tests__/magic-block-parser.test.js:231]: Having trailing commas here still ok?

File: __tests__/__snapshots__/flavored-compilers.test.js.snap (hunk 1 above)
Comment (kellyjosephprice) [reply __tests__/__snapshots__/flavored-compilers.test.js.snap:71]: Dang it.

File: __tests__/magic-block-parser.test.js (hunk 2 above)
Comment (kellyjosephprice) [reply __tests__/magic-block-parser.test.js:231]: I don't think so. It's never OK for `JSON.parse`, right?

Comment (erunion) [reply __tests__/magic-block-parser.test.js:231]: yeah that'll bust json.parse

**Review by erunion — APPROVED** [review]

**Comment by rafegoldberg** [comment]
## This PR was released!

🚀 Changes included in [<b>v6.37.0-next.1</b>](/readmeio/markdown/releases/tag/v6.37.0-next.1)
<br>

**Comment by rafegoldberg** [comment]
## This PR was released!

🚀 Changes included in [<b>v6.37.0</b>](/readmeio/markdown/releases/tag/v6.37.0)
<br>

**Pull request merged**